pipeline/src/convert_csv.ts

```diff
@@ -0,0 +1,77 @@
+import { program } from "commander";
+import util from "util";
+
+import FS from 'node:fs';
+import assert from 'node:assert';
+import {parse} from 'csv-parse/sync';
+
+import fs = require("node:fs/promises");
+import path = require("node:path");
+
+import { Table, filterTables, mergeTablesByModelNumber } from "./table_merger";
+import { APPLIANCE_TYPES } from "../../backend/schema/metadata";
+import { glob } from "glob";
+import { retrieveMetadata } from "./metadata";
+
+const SPECS_FILE_BASE = "../data/";
+const INPUT_SUBDIR = "reformatted/";
+const OUTPUT_SUBDIR = "";
+const RUNS = "runs/";
+
+program.parse();
+
+async function main() {
+// loop through the sub folders in the run folder
+const rfolder = path.join(SPECS_FILE_BASE, RUNS).replace(/\\/g, '/');
+FS.readdir(rfolder, async (err, topFolders) => {
+    if (err || !topFolders.length){
+        console.log(`No runs detected`);
+    }
+    else{
+        for (const file of topFolders){
+            // Try to open the "all_records.csv" file
+            const runFile = path.join(rfolder, file, "all_records.csv");
+            console.log(`Trying ${runFile}`);
+            FS.readFile(runFile, function (err, content) {
```

Comment:
A few notes on file reading and writing:

- It's a bit weird to have imports for both `node:fs` and `node:fs/promises`. Let's just pick one for the same file.
- if you use node:fs, I'd use the methods `readFileSync` and `writeFileSync` which avoid having to pass in a callback function. `pipeline/src/generate_metadata_files.ts` has some examples.
- if you use node:fs/promises, instead of passing in the callback function, which tends to lead to a lot of nesting, I'd use the `await` construct. Check out a file like `pipeline/src/reformat_tables.ts` for an example. If the file doesn't exist, it'll throw an error, but you can just let it kill the script since we can't do anything useful if the file is missing anyway.

Reply:
Changed all file read/writes to work on 'node:fs'